server/src/main/java/org/opensearch/rest/action/cat/RestTasksAction.java

```diff
@@ -137,6 +137,7 @@ protected Table getTableWithHeader(final RestRequest request) {
         // Task detailed info
         if (detailed) {
             table.addCell("description", "default:true;alias:desc;desc:task action");
+            table.addCell("resource_stats", "default:false;desc:resource consumption info of the task");
```

Comment:
There is computation involved for populating resource stats information and it can grow pretty large in future if we choose to add thread level information. Hence, adding this behind a control flag is better idea rather than adding this to the default task info. Currently detailed returns description, status and hence reusing the same ```detailed``` flag to control ```resource_stats``` . This not because of the task index compatibility.